datafusion/substrait/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -615,6 +621,91 @@ async fn extension_logical_plan() -> Result<()> {
     Ok(())
 }
 
+fn check_post_join_filters(rel: &Rel) -> Result<()> {
+    // search for target_rel and field value in proto
+    match &rel.rel_type {
```

Comment:
Looks like it might be helpful (eventually) do define `TreeNode` for `Rel` to implement walking efficiently.